homeassistant/components/fritz/common.py

```diff
@@ -285,3 +306,111 @@ def device_info(self):
             "model": self._fritzbox_tools.model,
             "sw_version": self._fritzbox_tools.sw_version,
         }
+
+
+class FritzBoxBaseSwitch:
+    """Fritz switch base class."""
+
+    def __init__(self, fritzbox_tools: FritzBoxTools, switch_info: SwitchInfo) -> None:
+        """Init Fritzbox port switch."""
+        self.fritzbox_tools: FritzBoxTools = fritzbox_tools
+
+        self._description = switch_info["description"]
+        self._friendly_name = switch_info["friendly_name"]
+        self._icon = switch_info["icon"]
+        self._type = switch_info["type"]
+        self._update: Callable = switch_info["callback_update"]
+        self._switch: Callable = switch_info["callback_switch"]
+
+        self._name = f"{self._friendly_name} {self._description}"
+        self._unique_id = (
+            f"{self.fritzbox_tools.unique_id}-{slugify(self._description)}"
+        )
```

Comment:
Wifi switches fail. I suppose that 5GHz and 2.4GHz switches get the same unique id (description matches):
```
2021-06-11 16:06:19 ERROR (MainThread) [homeassistant.components.switch] Platform fritz does not generate unique IDs. ID 3CA62F1BA198-wi_fi_rosefy already exists - ignoring switch.fritz_box_7530_wi_fi_rosefy
```

I think we should add a uniquer way to set the unique id.

Reply:
I think this happens because you have the same SSID for both 5 and 2.4.
We should skip the creation of the second, because in the end it's only 1 WiFi network.

Simone

Reply:
I disagree. Although it's one SSID it's two distinct networks. Disabeling one doesn't disable the other.

Reply:
Fixed with latest code tested together